ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/operators/SelectOperator.java

```diff
@@ -69,34 +69,32 @@ public void open() {
 
   @Override
   public Object next() {
-    row = (TableRow)child.next();
-    if (row == null) {
-      return null;
-    }
-
-    final GenericRow intermediate = PullPhysicalOperatorUtil.getIntermediateRow(
-        row, logicalNode.getAddAdditionalColumnsToIntermediateSchema());
-    final Function<TableRow, Optional<TableRow>> transformRow = row -> transformer.transform(
-        row.key(),
-        intermediate,
-        new PullProcessingContext(row.rowTime()))
-        .map(r -> {
-          if (logicalNode.isWindowed()) {
-            return WindowedRow.of(
-                logicalNode.getIntermediateSchema(),
-                ((WindowedRow) row).windowedKey(),
-                r,
-                row.rowTime());
-          }
-          return Row.of(logicalNode.getIntermediateSchema(), row.key(), r, row.rowTime());
-        });
-
-    Optional<TableRow> result = transformRow.apply(row);
+    Optional<TableRow> result = Optional.empty();
     while (result.equals(Optional.empty())) {
       row = (TableRow)child.next();
       if (row == null) {
         return null;
       }
+
+      final GenericRow intermediate = PullPhysicalOperatorUtil.getIntermediateRow(
+          row, logicalNode.getAddAdditionalColumnsToIntermediateSchema());
+      final Function<TableRow, Optional<TableRow>> transformRow = row -> {
```

Comment:
nit: I think it makes sense to just create this function outside of the while loop (or make it not a lambda) to avoid unnecessary function object creation (my preference, I don't see why this needs to be a lambda at all).

in fact, I think encapsulating it in a lambda is probably what caused the bug you described above in the first place (the `intermediate` was curried along with the function)!